test/backends.py

```diff
@@ -0,0 +1,217 @@
+__copyright__ = """
+Copyright (C) 2020 University of Illinois Board of Trustees
+Copyright (C) 2021 Esteban Cisneros
+"""
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+import cantera as ct
+import numpy as np
+import importlib
+import pathlib
+import typing
+
+import pyrometheus
+
+from collections.abc import Iterable
+from abc import abstractmethod
+
+try:
+    import jax
+    import jax.numpy as jnp
+except ImportError:
+    jnp = None
+else:
+    jax.config.update("jax_enable_x64", True)
+
+
+class Backend:
```

Comment:
The `Backend` classes need docstrings. Otherwise an elegant solution IMO.

Reply:
Added the docstrings.

Reply:
Thanks!